components/apprentice-status-indicator/apprentice-status-indicator.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from './apprentice-status-indicator.module.scss';
+
+const ApprenticeStatusIndicator = ({ status }) => {
+  if (!status) return null;
+
+  if (status === 'current') {
+    return (
+      <svg id="triangle" viewBox="0 0 100 100" className={styles['employee-status']}>
+        <polygon
+          points="50 15, 100 100, 0 100"
+          className={styles['current-employee-status']}
+          data-testid="current"
+        />
+      </svg>
+    );
+  }
+
+  if (status === 'previous') {
+    return (
+      <svg id="triangle" viewBox="0 0 100 100" className={styles['employee-status']}>
+        <polygon
+          points="50 15, 100 100, 0 100"
+          className={styles['past-employee-status']}
+          data-testid="previous"
+        />
+      </svg>
+    );
+  }
+
+  throw new Error(
+    `ApprenticeStatusIndicator: Expected "status" to not exist or be an expected value, but found ${status}`,
+  );
```

Comment:
I like what you're thinking here, but I would caution against this pattern for UI components. Typically, if your page components break, you want them to break "gracefully", meaning that the problem is localized to that component and the user still gets a good experience for the rest of the page. Usually what that means is that, if your component receives incorrect data, it either renders a friendly user message or nothing at all. In this case, we can allow the proptypes to handle this via console warnings in dev mode, and remove the explicit `throw`.

For a bit more nuance, in our particular case with our current ops model, this implementation wouldn't necessarily be a problem. With next.js/netlify, we are currently rendering everything statically at _build time_, so a data misconfiguration would break the build, rather than cause an error on the page. But typically we should build components in a way that they don't throw errors at runtime. We always want to _assume_ that components will be rendered on the client. Consider for example: the next apprentice class picks up where you left off, and we decide to move the data to something like [Contentful](https://www.contentful.com/), so that it can be managed outside of the codebase, fetched via an API call and rendered at runtime. If for some reason we get bad data for the indicator, throwing an error could break the page.